swift/Samples/ModalContainer/Sources/ModalContainerScreen.swift

```diff
@@ -0,0 +1,53 @@
+import WorkflowUI
+
+/// A `ModalContainerScreen` displays a base screen and optionally one or more modals on top of it.
+public struct ModalContainerScreen<BaseScreen: Screen>: Screen
+{
+
+    /// The base screen to show underneath any modally presented screens.
+    public var baseScreen: BaseScreen
+
+    /// Modally presented screens
+    public var modals: [Modal]
```

Comment:
Same here, can we make this `let`?

Reply:
good point. Will make both properties immutable.

Reply:
(Lurking in this PR)

I'd disagree here – generally always leave fields in structs as `var` unless there's a really good reason you want the field to never be set again after init – eg two fields are inherently related. Why? Structs are value types, so you can mutate your copy and it won't hurt anyone else. This makes reuse and mutation of your copy much easier – you can do it in place, versus having to re-call init on the type all the time.

Reply:
done.

Reply:
Sorry Kyle, i think i committed the changes as you were adding this comment but we ca still continue this discussion.

Reply:
No worries! Either is fine, just wanted to drop that thought in.

Reply:
I've been on the "make everything mutable" camp. But maybe I should re-consider that for value types. I remember discussing this with @AquaGeek a while ago.

Reply:
I often first make a modal container with its base in my `render`s then have code further down that checks which modals should be in it and append them as needed. It could certainly be reworked into one `ModalContainerScreen` initializer call, but I don’t think the immutability gets us any value here.